src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChange.java

```diff
@@ -51,102 +38,215 @@
  * To modify an existing feature: {@link ChangeType} is ADD, and the included reference needs to
  * contain the only the changed information related to that changed feature.
  * <p>
- * To remove an existing feature: {@link ChangeType} is REMOVE. The included reference's only
- * feature that needs to match the existing feature is the identifier.
+ * To remove an existing feature: {@link ChangeType} is REMOVE. The reference entity need only
+ * contain the identifier of the feature to remove.
+ * <p>
+ * For all {@link FeatureChange}s, you may indirectly include a reference to the before view of the
+ * entity using the {@link FeatureChange#add(AtlasEntity, Atlas)} and
+ * {@link FeatureChange#remove(AtlasEntity, Atlas)} methods. Providing the atlas context allows
+ * {@link FeatureChange} to perform more sophisticated merge logic.
  *
  * @author matthieun
+ * @author lcram
  */
 public class FeatureChange implements Located, Serializable
 {
     private static final long serialVersionUID = 9172045162819925515L;
-    private static final BinaryOperator<Map<String, String>> tagMerger = Maps::withMaps;
-    private static final BinaryOperator<Set<Long>> directReferenceMerger = Sets::withSets;
-    private static final BinaryOperator<SortedSet<Long>> directReferenceMergerSorted = Sets::withSortedSets;
-    private static final BinaryOperator<Set<Long>> directReferenceMergerLoose = (left,
-            right) -> Sets.withSets(false, left, right);
-    private static final BinaryOperator<RelationBean> relationBeanMerger = RelationBean::merge;
 
     private final ChangeType changeType;
-    private final AtlasEntity reference;
+    private AtlasEntity beforeView;
+    private final AtlasEntity afterView;
+
+    /**
+     * Create a new {@link ChangeType#ADD} {@link FeatureChange} with a given afterView. The
+     * afterView should be a {@link CompleteEntity} that specifies how the newly added or modified
```

Comment:
It could, technically. However, we disallow this in the constructor of `FeatureChange`.

```java
if (!(reference instanceof CompleteEntity))
{
    throw new CoreException(
        "FeatureChange requires CompleteEntity, found reference of type {}",
         reference.getClass().getName());
}
```

In fact, this was already the case before my updates - I simply notated this fact in the doc comment. I am in favor of enforcing `CompleteEntity`, since using a different entity type will almost always be due to a user error/misunderstanding/misuse.